client.go

```diff
@@ -1875,6 +1875,10 @@ func (c *Client) RecursivePushFile(container string, source string, target strin
 	}
 
 	sourceDir := filepath.Dir(source)
+	sourceLen := len(sourceDir)
+	if sourceDir == "." {
+		sourceLen--
+	}
 
```

Comment:
Why are you removing this? This has been added on purpose a couple of weeks ago.